src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/TouchstoneController.kt

```diff
@@ -71,21 +72,74 @@ class TouchstoneController(context: ControllerContext) : AbstractController(cont
         val source = context.params(":source-code")
         val type = context.params(":type-code")
         val gender = context.queryParams("gender")
-        return repo.getDemographicData(type, source, touchstone.id, gender?: "both")
+        val format = context.queryParams("format")
+
+        val splitData = repo.getDemographicData(type, source, touchstone.id, gender ?: "both")
+
+        return if (format == "wide")
+        {
+            SplitData<DemographicDataForTouchstone, WideDemographicRow>(splitData.structuredMetadata,
+                    getWideDemographicDatatable(splitData.tableData.data))
+        }
+        else
+        {
+            SplitData<DemographicDataForTouchstone, LongDemographicRow>(splitData.structuredMetadata,
+                    splitData.tableData)
+        }
     }
 
-    fun getDemographicData(context: ActionContext, repo: TouchstoneRepository): DataTable<DemographicRow>
+    private fun getWideDemographicDatatable(data: Iterable<LongDemographicRow>):
+            FlexibleDataTable<WideDemographicRow>
+    {
+        val groupedRows = data
+                .groupBy {
+                    "${it.countryCode} ${it.ageFrom} " +
+                            "${it.ageTo}"
```

Comment:
Might this be more readable as `.groupBy { Triple(it.countryCode, it.ageFrom, it.ageTo) }`, without the string interpolation gumph? Because `Triple` is a `data class`, it exhibits value semantics for `equals`, which I think it what you are shooting for here?